# 3 Comparing Consent-Asked to Zipcode-Frequency

In [1]:
import pandas as pd
import folium

In [2]:
consent_asked = pd.read_csv('data/consent_asked.csv')
zipcode_frequency = pd.read_csv('data/zipcode_frequency.csv')


In [3]:
consent_asked.head()

STOP_LOCATION_ZIP_CODE  TOTAL_STOPPED  ASK_FOR_CONSENT_FLG  PERCENT_ASKED
0                   10000              6                    0       0.000000
1                   10001             88                   14      15.909091
2                   10002            106                   10       9.433962
3                   10003             80                   10      12.500000
4                   10004              7                    1      14.285714

In [4]:
zipcode_frequency.head()

STOP_LOCATION_ZIP_CODE  FREQUENCY
0                   10000          6
1                   10001         92
2                   10002        109
3                   10003         82
4                   10004          7

The "frequency" numbers are slightly off from the "total asked" numbers since we dropped 3% from "frequency" where the notation showed neither yes or no for if consent was asked for. 

For these purposes, I'm going to use the "frequency" column instead of the "total asked" since that will include the data where the value of "(" was given for if someone consented. 

In [5]:
comparing_data = consent_asked.merge(zipcode_frequency, how = 'inner', on = "STOP_LOCATION_ZIP_CODE")

In [6]:
comparing_data.head()

STOP_LOCATION_ZIP_CODE  TOTAL_STOPPED  ASK_FOR_CONSENT_FLG  PERCENT_ASKED  \
0                   10000              6                    0       0.000000   
1                   10001             88                   14      15.909091   
2                   10002            106                   10       9.433962   
3                   10003             80                   10      12.500000   
4                   10004              7                    1      14.285714   

   FREQUENCY  
0          6  
1         92  
2        109  
3         82  
4          7

In [7]:
comparing_data.drop(columns = 'TOTAL_STOPPED')

STOP_LOCATION_ZIP_CODE  ASK_FOR_CONSENT_FLG  PERCENT_ASKED  FREQUENCY
0                     10000                    0       0.000000          6
1                     10001                   14      15.909091         92
2                     10002                   10       9.433962        109
3                     10003                   10      12.500000         82
4                     10004                    1      14.285714          7
..                      ...                  ...            ...        ...
173                   11436                    0       0.000000          5
174                   11691                    7      10.937500         66
175                   11692                    7      14.893617         48
176                   11693                    0       0.000000          9
177                   11694                    1       9.090909         11

[178 rows x 4 columns]

In [8]:
comparing_data['PERCENT_FREQUENCY'] = (comparing_data['FREQUENCY']/9544) * 100

In [9]:
comparing_data = comparing_data.sort_values('PERCENT_FREQUENCY', axis=0, ascending = False)

In [10]:
comparing_data = comparing_data.reset_index(drop = True)

In [11]:
top_25_frequency = comparing_data.head(25)
top_25_frequency

STOP_LOCATION_ZIP_CODE  TOTAL_STOPPED  ASK_FOR_CONSENT_FLG  PERCENT_ASKED  \
0                    10029            220                   25      11.363636   
1                    11207            169                   34      20.118343   
2                    10453            158                   25      15.822785   
3                    10458            146                   27      18.493151   
4                    10468            142                   19      13.380282   
5                    11212            150                   22      14.666667   
6                    10035            135                   20      14.814815   
7                    11206            130                   32      24.615385   
8                    11201            133                   27      20.300752   
9                    10456            127                   23      18.110236   
10                   11226            125                   22      17.600000   
11                   10027            115                   14      12.173913   
12                   10025            114                   17      14.912281   
13                   11233            107                   16      14.953271   
14                   11234            106                   25      23.584906   
15                   11220            107                    4       3.738318   
16                   10473            110                   14      12.727273   
17                   10002            106                   10       9.433962   
18                   10467            105                   11      10.476190   
19                   11213            105                   20      19.047619   
20                   10457            100                   15      15.000000   
21                   10460            102                   17      16.666667   
22                   11221             98                   22      22.448980   
23                   11216            100                   18      18.000000   
24                   10455            102                   15      14.705882   

    FREQUENCY  PERCENT_FREQUENCY  
0         227           2.378458  
1         176           1.844091  
2         159           1.665968  
3         154           1.613579  
4         154           1.613579  
5         152           1.592624  
6         140           1.466890  
7         135           1.414501  
8         134           1.404023  
9         128           1.341157  
10        128           1.341157  
11        120           1.257334  
12        118           1.236379  
13        114           1.194468  
14        114           1.194468  
15        113           1.183990  
16        113           1.183990  
17        109           1.142079  
18        108           1.131601  
19        107           1.121123  
20        106           1.110645  
21        104           1.089690  
22        104           1.089690  
23        103           1.079212  
24        102           1.068734

#### 10029: 
East Harlem, the most common stop and frisk zip code, percent asked is around 11%, (in between the first and second quartile, so slightly less than the mean%.) 

#### 11207: 
East New York, the second most common stop and frisk zip code, percent asked is around 20% (putting it in the 75 percentile of most consentual stop and frisks as well.) 

#### 11206: 
Bushwick, the seventh most common stop and frisk zip code, percent asked is around 25%, (also putting it in the 75% percentile of most consentual stop and frisks.) 

#### 11234: 
Flatlands/Bergen Beach, 14th most common stop and frisk zip code, percent asked is at 23% about. Similar to zip code 11206.

#### 11220:
Sunset Park, the 15th most common stop and frisk zip code, percent asked is the lowest% within the top 25 highest frequency zip code, at just under 4% asked. This is a very interesting case, since all the other zip codes in the top 25 most frequent zip codes have consent asked rates much higher by about 3 times at least. The next most consentual zip code in the top 25 is 10002 (The Lower East Side) with a consent asked rate of about 9%  


top_25_consent = comparing_data.sort_values('PERCENT_ASKED', axis=0, ascending = False).head(25)
top_25_consent

In [12]:
top_25_consent = comparing_data.sort_values('PERCENT_ASKED', axis=0, ascending = False)

In [13]:
top_25_consent.head()

STOP_LOCATION_ZIP_CODE  TOTAL_STOPPED  ASK_FOR_CONSENT_FLG  \
176                   11004              1                    1   
147                   10470             11                   10   
102                   10034             31                   12   
162                   11428              6                    2   
154                   11365              9                    3   

     PERCENT_ASKED  FREQUENCY  PERCENT_FREQUENCY  
176     100.000000          1           0.010478  
147      90.909091         11           0.115256  
102      38.709677         31           0.324811  
162      33.333333          6           0.062867  
154      33.333333          9           0.094300

In [14]:
top_25_consent_zips = top_25_consent['STOP_LOCATION_ZIP_CODE'].to_list()
top_25_frequency_zips = top_25_frequency['STOP_LOCATION_ZIP_CODE'].to_list()

top_25_consent_and_frequency = [x for x in top_25_consent_zips if x in top_25_frequency_zips]

In [15]:
top_25_consent_and_frequency

[11206,
 11234,
 11221,
 11201,
 11207,
 11213,
 10458,
 10456,
 11216,
 11226,
 10460,
 10453,
 10457,
 11233,
 10025,
 10035,
 10455,
 11212,
 10468,
 10473,
 10027,
 10029,
 10467,
 10002,
 11220]

#### 11206: 
Bushwick is on this list too, of top 25 most consentual zip codes for stop and frisks making it very interesting. 

#### 11234:
Sheepshead Bay is also on both lists. 

#### 11221: 
As are parts of Bedstuy and more of Bushwick. 

In [16]:
bottom_25_consent = comparing_data.sort_values('PERCENT_ASKED', axis=0, ascending = True).head(50)
bottom_25_consent

STOP_LOCATION_ZIP_CODE  TOTAL_STOPPED  ASK_FOR_CONSENT_FLG  \
177                   10168              1                    0   
103                   10306             31                    0   
110                   11377             29                    0   
111                   11210             27                    0   
122                   11412             20                    0   
81                    11232             42                    0   
130                   10475             17                    0   
131                   11378             17                    0   
138                   11357             13                    0   
139                   11416             13                    0   
151                   11693              9                    0   
152                   11362              9                    0   
142                   11358             12                    0   
156                   11411              8                    0   
175                   10153              1                    0   
174                   10121              1                    0   
173                   10048              2                    0   
172                   10069              2                    0   
171                   11360              3                    0   
153                   11418              9                    0   
106                   11355             30                    0   
167                   11436              5                    0   
166                   10006              5                    0   
163                   10000              6                    0   
158                   11426              7                    0   
168                   11415              4                    0   
70                    11375             47                    1   
98                    11364             34                    1   
99                    11204             31                    1   
104                   11379             31                    1   
15                    11220            107                    4   
74                    10312             47                    2   
116                   11361             23                    1   
50                    10012             67                    3   
117                   11103             22                    1   
83                    10065             40                    2   
123                   11429             20                    1   
124                   10075             19                    1   
125                   11105             19                    1   
129                   10308             18                    1   
95                    10017             34                    2   
105                   10022             30                    2   
43                    10036             75                    5   
31                    11385             88                    6   
108                   11374             29                    2   
101                   10030             29                    2   
79                    10128             41                    3   
33                    10026             82                    6   
87                    11420             39                    3   
67                    10014             51                    4   

     PERCENT_ASKED  FREQUENCY  PERCENT_FREQUENCY  
177       0.000000          1           0.010478  
103       0.000000         31           0.324811  
110       0.000000         29           0.303856  
111       0.000000         27           0.282900  
122       0.000000         21           0.220034  
81        0.000000         43           0.450545  
130       0.000000         17           0.178122  
131       0.000000         17           0.178122  
138       0.000000         14           0.146689  
139       0.000000         14           0.146689  
151       0.000000        

#### 11232:
This is Bush Terminal/Greenwood/Sunset park. Of the zip codes where consent was never asked for in 2020, this is the zip code where stop and frisks had the highest frequency, 

#### 11375:
Forrest Hills has a low consent asked rate too compared to it's stop and frisk frequency, similar to sunset park. 

#### 10306: 
Midland Beach has a surprising 0% asked rate for the 31 people stop and frisked there, compared to just next door, in 10309 were 22 people were stop and frisked. But when we look to try to see what percentage of those were asked, we see a percent close to 20% 

#### 11210: 
Flatlands had a low consent asked rate for the 30 about people who were stop and frisked there, in 2020, none were asked. 

In [17]:
bottom_25_consent_and_frequency = [x for x in bottom_25_consent if x in top_25_frequency_zips]
bottom_25_consent_and_frequency

[]

In [18]:
# comparing_data.to_excel('2020_NYPD_Stop-and-Frisk_consent-asked_rate.xlsx')